arangod/Replication2/StateMachines/Document/DocumentFollowerState.cpp

```diff
@@ -35,8 +35,9 @@ using namespace arangodb::replication2::replicated_state::document;
 
 DocumentFollowerState::DocumentFollowerState(
     std::unique_ptr<DocumentCore> core,
-    std::shared_ptr<IDocumentStateHandlersFactory> handlersFactory)
-    : _networkHandler(handlersFactory->createNetworkHandler(core->getGid())),
+    std::shared_ptr<IDocumentStateHandlersFactory> const& handlersFactory)
+    : shardId(core->getShardId()),
```

Comment:
The core is a guarded member. Because the `shardId` string view remains constant, I decided to store it like that, so I don't have acquire any lock to retrieve it.

Reply:
Ah, ok - that makes sense then!